app/controllers/posts_controller.rb

```diff
@@ -1,14 +1,64 @@
 class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+
   def index
-    @user = User.find(params[:user_id]) # Fetch the user by their ID
-    @posts = @user.posts # Fetch the posts related to the user
-    @recent_comments = @user.recent_comments # Fetch the recent comments for the user
-    @comments = @recent_comments
+    @user = User.find(params[:user_id])
+    @posts = @user.posts
   end
 
   def show
     @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
     @comments = @post.comments
+    @new_comment = Comment.new
+  end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = Post.new
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.build(post_params)
+
+    if @post.save
+      redirect_to user_post_path(@user, @post)
+    else
+      render :new
+    end
+  end
```

Comment:
- [ ] According to the requirements, a `Post` should be created on behalf of the `current_user`. However, here you're saving it with an author of `User.find(paramss[:user_id])`, which isn't necessarily the same as the `current_user`. Please consider fixing this! 🤓